src/libEth/Eth.cpp

```diff
@@ -73,28 +73,42 @@ Json::Value populateReceiptHelper(
   ret["r"] = GetR(sig);
   ret["s"] = GetS(sig);
 
+  switch (tx.GetVersionIdentifier()) {
+    case TRANSACTION_VERSION:
+      ret["type"] = "0x0"; // FIXME?
+      break;
+    case TRANSACTION_VERSION_ETH_LEGACY:
+      ret["type"] = "0x0";
+      break;
+    case TRANSACTION_VERSION_ETH_EIP_2930:
+      ret["type"] = "0x1";
+      break;
+    case TRANSACTION_VERSION_ETH_EIP_1559:
+      ret["type"] = "0x2";
+      break;
+    default:
+      LOG_GENERAL(WARNING, "Invalid transaction type");
+      break;
+  }
+
   return ret;
 }
 
-// Given a RLP message, parse out the fields and return a EthFields object
-EthFields parseRawTxFields(std::string const &message) {
+EthFields parseLegacyTransaction(zbytes asBytes) {
```

Comment:
Why is this being copied? unless you're planning to move this below (into RLP), better to pass a const reference.